h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamMojoModelBase.java

```diff
@@ -0,0 +1,170 @@
+package hex.genmodel.algos.gam;
+
+import hex.genmodel.ConverterFactoryProvidingModel;
+import hex.genmodel.GenModel;
+import hex.genmodel.MojoModel;
+import hex.genmodel.easy.CategoricalEncoder;
+import hex.genmodel.easy.EasyPredictModelWrapper;
+import hex.genmodel.easy.RowData;
+import hex.genmodel.easy.RowToRawDataConverter;
+import hex.genmodel.utils.ArrayUtils;
+import hex.genmodel.utils.DistributionFamily;
+import hex.genmodel.utils.LinkFunctionType;
+
+import java.util.Map;
+
+import static hex.genmodel.utils.ArrayUtils.nanArray;
+
+public abstract class GamMojoModelBase extends MojoModel implements ConverterFactoryProvidingModel {
+  public LinkFunctionType _link_function;
+  boolean _useAllFactorLevels;
+  int _cats;
+  int[] _catNAFills;
+  int[] _catOffsets;
+  int _nums;
+  int _numsCenter;
+  double[] _numNAFills;
+  double[] _numNAFillsCenter;
+  boolean _meanImputation;
+  double[] _beta;
+  double[] _beta_no_center;
+  double[] _beta_center;
+  double[][] _beta_multinomial;
+  double[][] _beta_multinomial_no_center; // coefficients not centered for multinomial/ordinal
+  double[][] _beta_multinomial_center; // coefficients not centered for multinomial/ordinal
+  DistributionFamily _family;
+  String[] _gam_columns;
+  int _num_gam_columns;
+  int[] _bs;
+  int[] _num_knots;
+  double[][] _knots;
+  double[][][] _binvD;
+  double[][][] _zTranspose;
+  String[][] _gamColNames;  // expanded gam column names
+  String[][] _gamColNamesCenter;
+  String[] _names_no_centering; // column names of features with no centering
+  int _totFeatureSize; // Gam Algo predictors numbers that include: predictors, expanded gam columns no centered
+  int _betaSizePerClass;
+  int _betaCenterSizePerClass;
+  double _tweedieLinkPower;
+  double[][] _basisVals; // store basis values array for each gam column
+  double[][] _hj;   // difference between knot values
+  int _numExpandedGamCols; // number of expanded gam columns
+  int _lastClass;
+  
+  public int getTotFeatureSize() { return _totFeatureSize;}
+  public String[] getNames() { return _names; }  // get column names centered
```

Comment:
can be removed, overrides already existing public method

Reply:
Done